core/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -290,11 +294,12 @@ public QuerySearchResultProvider executeQueryPhase(ShardSearchRequest request) t
         }
     }
 
-    public ScrollQuerySearchResult executeQueryPhase(InternalScrollSearchRequest request) {
+    public ScrollQuerySearchResult executeQueryPhase(InternalScrollSearchRequest request, SearchTask task) {
         final SearchContext context = findContext(request.id());
         SearchOperationListener operationListener = context.indexShard().getSearchOperationListener();
         context.incRef();
         try {
+            context.setTask(task);
```

Comment:
Oooh! Yeah! That makes sense - scrolls contexts will outlast that task that spawns them and need to get a new task.

I haven't read the whole PR yet, but I wonder if we should clear task from the context between requests? If we don't then we maintain a reference to it after the task manager and that seems a bit weird. Changes to it wouldn't be useful to anyone.